lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -817,20 +836,12 @@ sub _generate_job_setting {
     }
 
     # TEST is mandatory, so populate with TestSuite settings.
-    my $test_suites = $schema->resultset('TestSuites')->search(
-        {
-            name => $args->{TEST},
-        });
-
-    if (my $test_suite = $test_suites->next) {
-        my %test_suite_setting = map { $_->key => $_->value } $test_suite->settings;
-
+    if (my %test_suite_setting = %{_test_suite_settings($schema, $args->{TEST})}) {
```

Comment:
Why are you creating a copy of the hash here?

Reply:
you mean I should work on the reference? Ok

Reply:
Yes, that's the alternative to the copy.